src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,145 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/Memory.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(std::move(generators))
+    , mIndex(0)
+  {
+    for (std::size_t i = 0; i < mGenerators.size(); ++i)
+    {
+      if (!mGenerators[i])
+      {
+        std::stringstream msg;
+        msg << "Generator " << i << " is nullptr.";
+        throw std::invalid_argument(msg.str());
+      }
+    }
+
+#ifndef NDEBUG
+    for (const auto& generator : mGenerators)
+      assert(generator->getStateSpace() == mStateSpace);
+#endif
+  }
+
+  // Documentation inherited
+  statespace::StateSpacePtr getStateSpace() const override
+  {
+    return mStateSpace;
+  }
+
+  // Documentation inherited
+  bool sample(statespace::StateSpace::State* state) override
+  {
+    while (mIndex < mGenerators.size())
+    {
+      if (mGenerators[mIndex]->sample(state))
+        return true;
+
+      mIndex++;
+    }
+    return false;
+  }
+
+  // Documentation inherited
+  int getNumSamples() const override
+  {
+    int numSamples = 0;
+    for (std::size_t i = mIndex; i < mGenerators.size(); ++i)
+    {
+      if (mGenerators[i]->getNumSamples() == NO_LIMIT)
+        return NO_LIMIT;
+
+      numSamples += mGenerators[i]->getNumSamples();
+    }
+    return numSamples;
+  }
+
+  // Documentation inherited
```

Comment:
Nit: `//==============================================================================`, not `// Documentation inherited`.